src/mappings/poolCommitter.ts

```diff
@@ -177,17 +184,117 @@ export function executedCommitsForInterval(event: ExecutedCommitsForInterval): v
+			let aggregateBalancesEntity = UserAggregateBalances.load(aggregateBalanceId);
+
+			if (!aggregateBalancesEntity) {
+				aggregateBalancesEntity = initUserAggregateBalance(poolId, trader)
+			}
+			const aggregateBalances = poolCommitterInstance.getAggregateBalance(trader as Address);
+
+			const shortTokenDiff = aggregateBalances.shortTokens.minus(aggregateBalancesEntity.shortTokenHolding)
+			const longTokenDiff = aggregateBalances.longTokens.minus(aggregateBalancesEntity.longTokenHolding)
+
+
+			// if aggregateBalances go to 0 this will be negative.
+			// mathematically I think without this check the weighted avg would still work
+			if (shortTokenDiff.le(BigInt.fromI32(0))) {
+				// this means no change in balances or 
+				aggregateBalancesEntity.shortTokenAvgBuyIn = BigInt.fromI32(0)
+			} else {
+				aggregateBalancesEntity.shortTokenAvgBuyIn = BigInt.fromString(calcWeightedAverage(
+					[aggregateBalancesEntity.shortTokenAvgBuyIn, shortPrice], // values 
+					[aggregateBalancesEntity.shortTokenHolding, shortTokenDiff], // weights
+				).toString())
+			}
+
+			if (longTokenDiff.le(BigInt.fromI32(0))) {
+				aggregateBalancesEntity.longTokenAvgBuyIn = BigInt.fromI32(0)
+			} else {
+				aggregateBalancesEntity.longTokenAvgBuyIn = BigInt.fromString(calcWeightedAverage(
+					[aggregateBalancesEntity.longTokenAvgBuyIn, longPrice], // values 
+					[aggregateBalancesEntity.longTokenHolding, longTokenDiff], // weights
+				).toString())
+			}
+
+			aggregateBalancesEntity.longTokenHolding = aggregateBalances.longTokens;
+			aggregateBalancesEntity.shortTokenHolding = aggregateBalances.shortTokens;
+			aggregateBalancesEntity.settlementTokenHolding = aggregateBalances.settlementTokens;
 
-			commit.save()
+			aggregateBalancesEntity.save();
 		}
 
+
 		// remove the pending commits from the store
 		store.remove('PendingCommitsByPoolCommitterAndInterval', relevantPendingCommitsId)
 	}
-}
\ No newline at end of file
+}
+
+
+export function claim(event: Claim): void {
+	let leveragedPoolByPoolCommitter = LeveragedPoolByPoolCommitter.load(event.address.toHexString());
+	if(!leveragedPoolByPoolCommitter) {
+		throw new Error('LeveragedPoolByPoolCommitter not set when handling new commit')
+	}
+
+	let pool = LeveragedPoolEntity.load(leveragedPoolByPoolCommitter.pool.toHexString());
+
+
+	// let poolCommitterInstance = PoolCommitter.bind(event.address);
+
+	if(!pool) {
+		throw new Error('LeveragedPool not found when handling new commit')
+	}
+
+	const aggregateBalanceId = pool.id.toString()+'-'+event.params.user.toHexString();
+	let aggregateBalance = UserAggregateBalances.load(aggregateBalanceId)
+
+	if (!aggregateBalance) {
+		aggregateBalance = initUserAggregateBalance(pool.id, event.params.user);
+	}
+
+	aggregateBalance.longTokenHolding = BigInt.fromI32(0);
+	aggregateBalance.shortTokenHolding = BigInt.fromI32(0);
+	aggregateBalance.settlementTokenHolding = BigInt.fromI32(0);
+
+
+	aggregateBalance.save();
+}
```

Comment:
I don't we need this?

Reply:
otherwise it will display a balance if they have claimed and not committed again